src/middleware/auth.test.ts

```diff
@@ -0,0 +1,183 @@
+
+    nock('https://www.googleapis.com:443', {"encodedQueryParams": true})
+      .get('/oauth2/v3/userinfo')
+      .query({"access_token":  "test"})
+      .reply(200, {"sub": "105658610279018384397", "name": "Kevin Schoonover", "given_name": "Kevin", "family_name": "Schoonover", "profile": "https://plus.google.com/105658610279018384397", "picture": "https://lh3.googleusercontent.com/-6q2444XuqBQ/AAAAAAAAAAI/AAAAAAAAAAA/AAN31DVOVtTyYweEGML7SvkzEgU1F0o9zg/mo/photo.jpg", "email": "ksyh3@mst.edu", "email_verified": true, "locale": "en", "hd": "mst.edu"});
+    request = agent(server);
+    done();
+  });
+  afterEach(async () => {
+    nock.cleanAll();
+    jest.clearAllMocks();
+    await server.close();
+  });
+
+  describe("Valid authentication", async () => {
+    it("should login user if user exists in database", async (done) => {
+      Account.mockImplementation(() => {
+        return {
+          save: () => {},
+        };
+      });
+      Account.findOne = jest.fn().mockReturnValue({id: "test", firstName: "Kevin", lastName: "Schoonover", email: "ksyh3@mst.ed"});
+      await login_user(request);
+      await verify_login_redirect(request, "/authenticated/");
+      done();
+    });
+
+    it("should login user if user does not exist in database", async (done) => {
+      let user;
+      Account.mockImplementation(() => {
+        return {
+          save() {
+            this.id = "test";
+            user = this;
+          },
+        };
+      });
+      Account.findOne = jest.fn()
+        .mockReturnValueOnce(undefined)
+        .mockReturnValue({"id": "test"});
+
+      await login_user(request);
+      expect(user.firstName).toBeDefined();
+      expect(user.lastName).toBeDefined();
+      expect(user.email).toBeDefined();
+      await verify_login_redirect(request, "/authenticated/");
+      done();
+    });
+  });
+
+  describe("Invalid authentication", async () => {
+    it("should fail to reauthenticate if deserialization fails", async (done) => {
+      let user;
+      Account.mockImplementation(() => {
+        return {
+          save() {
+            this.id = "test";
+            user = this;
+          },
+        };
+      });
+      Account.findOne = jest.fn()
+        .mockReturnValueOnce(undefined)
```

Comment:
Do we need both mockReturnValueOnce() and mockReturnValue() if they both are the same return?